application/src/main/java/org/togetherjava/tjbot/features/moderation/history/PurgeHistoryCommand.java

```diff
@@ -0,0 +1,123 @@
+package org.togetherjava.tjbot.features.moderation.history;
+
+import static org.togetherjava.tjbot.db.generated.Tables.MESSAGE_HISTORY;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.unions.MessageChannelUnion;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.SlashCommandAdapter;
+
+public class PurgeHistoryCommand extends SlashCommandAdapter {
+
+    private static final String USER_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+    private static final String COMMAND_NAME = "purge";
+    private final Database database;
+
+    /**
+     * Creates an instance of the command.
+     *
+     * @param database the database to retrieve message records of a target user
+     */
+
+    public PurgeHistoryCommand(Database database) {
+        super(COMMAND_NAME, "delete user message history upto an hour", CommandVisibility.GUILD);
```

Comment:
the command only purges within the given channel, no? not clear from the description.
also, to prevent confusion, id actually rename the command then. perhaps to `purge-in-channel`

Reply:
yes, that's correct. Will do.